lib/reek/cli/options.rb

```diff
@@ -136,6 +139,9 @@ def set_up_verbosity_options
                   'Show link to related wiki page for each smell (default: true)') do |show_links|
           self.show_links = show_links
         end
+        parser.on('-P', '--[no-]progress', 'Show progress of each file (default: true)') do |show_progress|
```

Comment:
I debated whether or not this could be in it's own method (like colored output), but it felt well placed within the verbosity options.